### ISSUE TITLE
HTTP2: System.OverflowException: Arithmetic operation resulted in an overflow.

### ISSUE THREAD

**Issue by JamesNK** [dotnet/corefx#37445]
An error is raised from the gRPC interop tests with HttpClient.SendAsync:

```
System.OverflowException: Arithmetic operation resulted in an overflow.
   at System.Net.Http.CreditManager.AdjustCredit(Int32 amount)
   at System.Net.Http.Http2Connection.RemoveStream(Http2Stream http2Stream)
   at System.Net.Http.Http2Connection.Http2Stream.Cancel()
   at System.Net.Http.Http2Connection.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncUnbuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Grpc.NetCore.HttpClient.Internal.GrpcCall`2.SendAsync(HttpClient client, HttpRequestMessage message) in /var/local/git/grpc-dotnet/src/Grpc.NetCore.HttpClient/Internal/GrpcCall.cs:line 349
   at Grpc.NetCore.HttpClient.Internal.GrpcCall`2.GetResponseAsync() in /var/local/git/grpc-dotnet/src/Grpc.NetCore.HttpClient/Internal/GrpcCall.cs:line 250
```

https://source.cloud.google.com/results/invocations/b97f0bdb-5797-4f24-b9b1-54b30875a5b7/targets/github%2Fgrpc%2Finterop_test/tests;group=tests;test=cloud_to_cloud:aspnetcore:c%2B%2B_server:cancel_after_begin:tls;row=1

**Comment by stephentoub** [comment]
What is the test doing? How do we repro?

**Comment by JamesNK** [comment]
That's going to be tough. The error happens against non-.NET servers in the gRPC interop environment, but not against Kestrel on my computer.

I've looked through the interop tests and this error only comes up for the cancel_after_begin test. In this test scenario content is being streamed to the server via HttpRequestMessage.HttpContent and the request is cancelled via SendAsync's cancellation token.

**Comment by JamesNK** [comment]
Wireshark captures of success against ASP.NET Core and failure against C Core: [wireshark-cancel_after_begin.zip](https://github.com/dotnet/corefx/files/3145142/wireshark-cancel_after_begin.zip)

**Comment by JamesNK** [comment]
Repo:
1. git clone https://github.com/JamesNK/grpc-dotnet.git
2. git checkout jamesnk/nativeinteropserver
3. dotnet run -p testassets\InteropTestsNativeWebsite
4. dotnet run -p testassets\InteropTestsClient --server_port 8070 --test_case cancel_after_begin

**Comment by wfurt** [comment]
I'm able to reproduce it. Since it involves RemoveStream(), it may be fixed by changes for #37455 
I'll do more testing when we have fix in place but the issue is understood.

**Comment by wfurt** [comment]
with updated code:
```
furt@net-dale:/mnt/github/grpc-dotnet/testassets/InteropTestsClient$ ~/dotnet-3.0.x/dotnet run --server_port 8070 --test_case cancel_after_begin
Use TLS: False
Use Test CA: False
Client type: httpclient
Server host: localhost
Server port: 8070
running cancel_after_begin
Passed!
```
